### ISSUE TITLE
upsert method doesn't trigger any hooks for mysql adapter

### ISSUE THREAD

**Issue by bahung1221** [1602/jugglingdb#461]
Hi,

I found that the upsert method won't trigger any hook if the adapter support updateOrCreate method (mysql in my case). I debugged and found the reason in below code:

```javascript
if (this.schema.adapter.updateOrCreate) {
        const inst = new Model(data);
        this.schema.callAdapter('updateOrCreate', Model.modelName, stripUndefined(inst.toObject(true)), (err, data) => {
            if (data) {
                return inst.reload(callback);
            }
            callback(err, null);
        });
    } else {
    ...
}
```

I think it should trigger *save/update* hooks like other methods,
If you need, i can create a PR for it,

Thank you very much for a nice job!

**Comment by 1602** [comment]
Yep, that seems like a missing call, I'd appreciate PR, thank you!

**Comment by bahung1221** [comment]
Hi,

I fixed it by make it work same behavior for all adapter,
It will make the workflow of upsert (createOrUpdate) method more consistency.
I think it is the best way because if we call updateOrCreate method of adapter, we can't know that it was create or update.
Pull request: https://github.com/1602/jugglingdb/pull/462

Thank you

**Comment by 1602** [comment]
Merged. One test is failing, though. I'll try to find some time to investigate why.
